src/m365/spo/commands/folder/folder-sharinglink-list.ts

```diff
@@ -0,0 +1,144 @@
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'scope', 'roles', 'link'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+    this.#initValidators();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        folderUrl: typeof args.options.folderUrl !== 'undefined',
+        folderId: typeof args.options.folderId !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '-u, --webUrl <webUrl>' },
+      { option: '--folderUrl [folderUrl]' },
+      { option: '--folderId [folderId]' },
+      {
+        option: '-s, --scope [scope]',
+        autocomplete: this.allowedScopes
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['folderUrl', 'folderId'] }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.folderId && !validation.isValidGuid(args.options.folderId)) {
+          return `${args.options.folderId} is not a valid GUID`;
+        }
+
+        if (args.options.scope && this.allowedScopes.indexOf(args.options.scope) === -1) {
+          return `'${args.options.scope}' is not a valid scope. Allowed values are: ${this.allowedScopes.join(',')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('webUrl', 'folderUrl', 'folderId', 'scope');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Retrieving sharing links for folder ${args.options.folderId || args.options.folderUrl}...`);
+    }
+
+    try {
+      const relFolderUrl: string = await spo.getFolderServerRelativeUrl(args.options.webUrl, args.options.folderUrl, args.options.folderId);
+      const absoluteFolderUrl: string = urlUtil.getAbsoluteUrl(args.options.webUrl, relFolderUrl);
+      const folderUrl: URL = new URL(absoluteFolderUrl);
+
+      const siteId: string = await spo.getSiteId(args.options.webUrl);
+      const drive: Drive = await driveUtil.getDriveByUrl(siteId, folderUrl);
+      const itemId: string = await driveUtil.getDriveItemId(drive, folderUrl);
+
+      let requestUrl = `https://graph.microsoft.com/v1.0/drives/${drive.id}/items/${itemId}/permissions?$filter=Link ne null`;
+      if (args.options.scope) {
+        requestUrl += ` and Link/Scope eq '${args.options.scope}'`;
+      }
+
+      const sharingLinks = await odata.getAllItems<any>(requestUrl);
+
```

Comment:
As discussed in the doc file, let's remove the grantedIdentities from the object here.